mxcube3/core/qutils.py

```diff
@@ -1851,16 +1852,18 @@ def queue_stop():
             qe.stop()
         except Exception as ex:
             print str(ex)
-
-        logging.getLogger('user_level_log').info(
-            'Queue execution was aborted, ' + str(qe.get_data_model()))
-
-        blcontrol.queue.queue_hwobj.set_pause(False)
-        # the next two is to avoid repeating the task
-        # TODO: if you now run the queue it will be enabled and run
-        qe.get_data_model().set_executed(True)
-        qe.get_data_model().set_enabled(False)
-        qe._execution_failed = True
+        
+        try:
+            logging.getLogger('user_level_log').info(
```

Comment:
What is this try handling ?, and would it not be better to write any exception to log ?

Reply:
removed the unnecessary log.
It can happen that one tries to stop an empty queue (current_entry none), then you will see the red error bar in the interface.

Reply:
Hm, I see that's really strange because we can only stop a queue that is running and the queue can only be running if contains something, right ?

In anyways, I guess its good to handle the scenario jut in case. I am however wondering if its not a effect of something else going wrong that you are seeing ?

Reply:
When mounting a sample (no queue operation) the cancel button on the pop up modal is linked to `sendStopQueue`, this method will fail if the queue is empty, which is the case if you simply mount a sample.
I changed a little.

The alternative would be to send abort to the SC instead of stopping the queue, but I am not sure if that is a good solution...

Reply:
I see, okey I think the sendStopQueue is there in case we are running the queue in pipeline mode.

I think this is fine, for now but its maybe worth looking into what the desired action of cancel would be.

Reply:
what do you think ?

Reply:
I could see that if we are cancelling via SC in operation modal, one should send a abortSC and then stopQueue if the queue was running. But, letting the users abort the SC operation may cause extra trouble (e.g. stop in the middle of a loading and the they would need to recover from there)

Reply:
Yes, that's also the reason for why are using stopQueue today and that we a while back wanted to remove the cancel possibility entirely. Its good if its possible but it does often cause other maybe more serious problems. Although its finally an "issue" of the sample changer abort function it often causes us undesired side effects

Reply:
Lets leave it like it is for the moment, Ill re-discuss it with the people here to see if any other ideas have surfaced.